src/main/java/com/exadel/discount/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package com.exadel.discount.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity<ExceptionDetails> handleException(NotFoundException notFoundException) {
+        ExceptionDetails exceptionDetails = new ExceptionDetails(notFoundException.getMessage());
+        return new ResponseEntity<>(exceptionDetails, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ExceptionDetails> handleException(MethodArgumentNotValidException notValidException) {
+        String message = notValidException.getBindingResult().getAllErrors().get(0).getDefaultMessage();
```

Comment:
1. getting first element is always tricky, because could cause exception without checking
2. Why are you getting just first validation error? why not all?

Reply:
+